### PR TITLE
PR for #33: SQL peer review exercise - Erick

### PR DESCRIPTION
**Pull request by sambock27**
Hello @Erick11293, There is one file for you to review as part of this PR:

- [ ] [`patient_demographics`](https://github.com/dchan-lab/practice/blob/issue33_sql_peer_review/build/code/PatientDemographics.sql)

I've assigned you as the reviewer for this PR. Please peer review the file linked above. Use our best practices for SQL code review [page](https://github.com/dchan-lab/dchanlab/wiki/Best-Practices-for-SQL-Code-Review) as a guide. As a reviewer, you aren't supposed to make any changes to the script.

Let me know if you have any questions!

### CODE REVIEW

File: build/code/PatientDemographics.sql [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:1]: There is no preamble comment block

File: build/code/PatientDemographics.sql [hunk 2]
```diff
@@ -0,0 +1,58 @@
+
+USE [ORD_Chan_201606104D]
+GO
+
+-- Patient Race
+IF Object_Id('tempdb..#PatientRace') IS NOT NULL
+	DROP TABLE #PatientRace
+
+SELECT *
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:9]: Here is a massive query, but according to the wiki the variables should be specified

File: build/code/PatientDemographics.sql [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+USE [ORD_Chan_201606104D]
+GO
+
+-- Patient Race
+IF Object_Id('tempdb..#PatientRace') IS NOT NULL
+	DROP TABLE #PatientRace
+
+SELECT *
+INTO #PatientRace
+	FROM [Src].[PatSub_PatientRace]
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:11]: Unnecessary indent

File: build/code/PatientDemographics.sql [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+GROUP BY PatientSID;
+
+
+-- Patient Ethnicity
+IF Object_Id('tempdb..#PatientEthnicity') IS NOT NULL
+	DROP TABLE #PatientEthnicity
+
+SELECT PatientSID, MAX(CASE WHEN Ethnicity IN ('HISPANIC OR LATINO') THEN 1 ELSE 0 END) AS Ethnicity_Hispanic
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:19]: This line should be split in different lines

File: build/code/PatientDemographics.sql [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	MAX(V.DOD) AS DateOfDeath,
+	MAX(CASE WHEN V.SEX = 'M' THEN 1 ELSE 0 END) AS Sex_M,
+	-- Race and ethnicity
+	MAX(R.Race_API) AS Race_API, 
+	MAX(R.Race_Black) AS Race_Black,  
+	MAX(R.Race_WhiteNotHispanic) AS Race_WhiteNotHispanic,
+	MAX(E.Ethnicity_Hispanic) AS Ethnicity_Hispanic
+INTO [Dflt].[NonVA_PatientDemographics_15]
+FROM (
+	SELECT DISTINCT ScrSSN
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:42]: According to the wiki the SELECT DISTINCT should be avoided

File: build/code/PatientDemographics.sql [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	FROM [Dflt].[NonVA_P1_Ambulances_15]
+) AS A
+INNER JOIN [Src].[SPatient_SPatient] AS CX
+	ON CONVERT(int,A.ScrSSN) = CONVERT(int,CX.ScrSSN)
+LEFT JOIN #PatientEthnicity AS E
+ON CX.PatientSID = E.PatientSID
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:48]: This line should be indent

File: build/code/PatientDemographics.sql [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+LEFT JOIN #PatientRace AS R
+	ON CX.PatientSID = R.PatientSID
+LEFT JOIN [Src].[VitalStatus_Mini] AS V
+	ON CX.ScrSSN=V.SCRSSN
+LEFT JOIN [Src].[Inpat_Inpatient] AS I
+	ON R.PatientSID = I.PatientSID
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:54]: There is no use of this table as a source of variables

File: build/code/PatientDemographics.sql [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+WHERE CX.ScrSSN IS NOT NULL AND NOT CX.ScrSSN = '*Unknown at this time*' AND NOT CX.scrSSN = '*Missing*'
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:55]: Each clause in this line should be in different lines

File: build/code/PatientDemographics.sql [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+GROUP BY CX.ScrSSN;
+
+
```
Comment (Erick11293) [inline build/code/PatientDemographics.sql:58]: The compression is missing

**Comment by Erick11293** [comment]
Hi @sambock27 , that are all the comments I have. Let me know if there any that I missed.

File: build/code/PatientDemographics.sql (hunk 1 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:1]: BINGO

File: build/code/PatientDemographics.sql (hunk 3 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:11]: Yessss

File: build/code/PatientDemographics.sql (hunk 4 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:19]: Oh yeah!

File: build/code/PatientDemographics.sql (hunk 5 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:42]: This a good point, and one that I had not caught before. I think in this case, it is ok. But, you have the right idea about identifying parts of the code that could be inefficient or wrong during a PR. If this were a real PR, you would still highlight it and confirm that is the right idea.

The purpose of the `select distinct` here is to create a crosswalk of scrambled social security numbers to use later on in the pull. Thus, it is ok.

File: build/code/PatientDemographics.sql (hunk 6 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:48]: Yeppers

File: build/code/PatientDemographics.sql (hunk 7 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:54]: YES! ding ding ding.

File: build/code/PatientDemographics.sql (hunk 8 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:55]: yes yes yes. YES

File: build/code/PatientDemographics.sql (hunk 9 above)
Comment (sambock27) [reply build/code/PatientDemographics.sql:58]: great catch

**Comment by sambock27** [comment]
Great work on this PR, @Erick11293! Those were all great catches. The only thing I can think of adding is a comment encouraging them to demonstrate that the data are unique at some level. Other than that, I think you caught it all.

Feel free to close this PR (but don't merge it to main, and don't delete the branch).

**Pull request closed without merging**